core/slice.go

```diff
@@ -62,24 +63,27 @@ type Slice struct {
 	bestPhKey common.Hash
 	phCache   map[common.Hash]types.PendingHeader
 
-	validator  Validator // Block and state validator interface
-	tpsHistory *lru.Cache
-	subTps     []uint32
-	latestTps  uint32
+	validator     Validator // Block and state validator interface
+	tpsHistory    *lru.Cache
+	subTps        []uint32
+	latestTps     uint32
+	appendHistory *lru.Cache
 }
 
 func NewSlice(db ethdb.Database, config *Config, txConfig *TxPoolConfig, isLocalBlock func(block *types.Header) bool, chainConfig *params.ChainConfig, domClientUrl string, subClientUrls []string, engine consensus.Engine, cacheConfig *CacheConfig, vmConfig vm.Config, genesis *Genesis) (*Slice, error) {
 	nodeCtx := common.NodeLocation.Context()
 	tpsLru, _ := lru.New(10)
+	appendTimeCache, _ := lru.New(10)
 	sl := &Slice{
-		config:     chainConfig,
-		engine:     engine,
-		sliceDb:    db,
-		domUrl:     domClientUrl,
-		quit:       make(chan struct{}),
-		tpsHistory: tpsLru,
-		subTps:     make([]uint32, 3),
-		latestTps:  0,
+		config:        chainConfig,
+		engine:        engine,
+		sliceDb:       db,
+		domUrl:        domClientUrl,
+		quit:          make(chan struct{}),
+		tpsHistory:    tpsLru,
+		subTps:        make([]uint32, 3),
+		latestTps:     0,
```

Comment:
Why does TPS stuff come in this commit? It should have already come in Connor's TPS commits before your PR.